server/game/gamesteps/plot/selectplotprompt.js

```diff
@@ -25,6 +29,10 @@ class SelectPlotPrompt extends AllPlayerPrompt {
 
     onMenuCommand(player, arg) {
         if(arg === 'changeplot') {
+            if(player.mustRevealPlot) {
```

Comment:
Preventing them from changing plots is good, also removing the Change Plot button entirely if they aren't allowed to change would be better UX.